packages/orama/src/components/facets.ts

```diff
@@ -56,23 +57,60 @@ export async function getFacets(
     for (const facet of facetKeys) {
       const facetValue = facet.includes('.')
         ? (await getNested<string>(doc!, facet))!
-        : (doc![facet] as number | boolean)
+        : (doc![facet] as SearchableType | Array<SearchableType>)
 
       // Range facets based on numbers
       if (properties[facet] === 'number') {
         for (const range of (facetsConfig[facet] as NumberFacetDefinition).ranges) {
-          if ((facetValue as number) >= range.from && (facetValue as number) <= range.to) {
-            if (facets[facet].values[`${range.from}-${range.to}`] === undefined) {
-              facets[facet].values[`${range.from}-${range.to}`] = 1
-            } else {
-              facets[facet].values[`${range.from}-${range.to}`]++
+          if (Array.isArray(facetValue)) {
+            const alreadyIncremeted = new Set()
+            for (let i = 0; i < facetValue.length; i++) {
```

Comment:
As a general rule, replace

```js
for (let i = 0; i < facetValue.length; i++) {
```

with 

```js
const facetValueLength = facetValue.length
for (let i = 0; i < facetValueLength; i++) {
```

to avoid calculating the length of an array on every iteration